packages/context/src/inject.ts

```diff
@@ -124,7 +124,7 @@ export function inject(
   }
   return function markParameterOrPropertyAsInjected(
     target: Object,
-    member: string,
+    member?: string,
```

Comment:
IIRC, we can use `''` as `member` for constructor parameters.

Reply:
I am reluctant to set `member= ''` because that's different from what TypeScript does.

Here is the code emitted by `tsc` for a `TodoRepository`:

```js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
  // [removed for brevity]
};

// [removed for brevity]

let TodoRepository = class TodoRepository 
  extends repository_1.DefaultCrudRepository 
{
  constructor(dataSource) {
    super(models_1.Todo, dataSource);
  }
};
TodoRepository = __decorate([
  __param(0, core_1.inject('datasources.db')),
  __metadata("design:paramtypes", [repository_1.juggler.DataSource])
], TodoRepository);
```

Notice that `__decorate` is called with `key` set to `undefined`.

Reply:
See https://github.com/strongloop/loopback-next/pull/3634

Reply:
And also https://github.com/microsoft/TypeScript/issues/33260